pandas/core/indexes/base.py

```diff
@@ -3902,6 +3903,15 @@ def reindex(
                         )
                     indexer, _ = self.get_indexer_non_unique(target)
 
+                if not self.is_unique:
```

Comment:
isn't this the same check as L3892?

Reply:
no, self.is_unique is _slightly_ different from self._index_as_unique